src/cls/queue/cls_queue_src.cc

```diff
@@ -67,12 +67,12 @@ int queue_read_head(cls_method_context_t hctx, cls_queue_head& head)
     return -EINVAL;
   }
 
-  uint8_t decoded_head_size = sizeof(uint64_t) + sizeof(uint16_t);
+  constexpr uint8_t decoded_head_size = sizeof(uint64_t) + sizeof(uint16_t);
   if (encoded_len > (chunk_size - decoded_head_size)) {
+    start_offset = chunk_size;
```

Comment:
@pritha-srivastava this is a fix to the cls_queue code, when head is larger than 1024

Reply:
@yuvalif : sorry but this is not correct, while reading the head, we first read the QUEUE_HEAD_START, and then the encoded size of head, and then based on the encoded size if it is greater than 1k we read the head, and here we read the data again starting from offset (leaving head start and encoded length) so that it is correctly decoded to cls_queue_head. Here 'it' will automatically be incremented by decoded_head_size in bl_head. What issue are you exactly facing for urgent data? (Both QUEUE_START_HEAD and encoded size are not part of cls_queue_head)